utils/__tests__/gridLayout.spec.ts

```diff
@@ -0,0 +1,386 @@
+import { beforeEach, describe, expect, it } from 'vitest'
+import { toGridLayoutItems, toLSP27TheGrid } from '../gridLayout'
+
+const TEST_LSP27_THE_GRID: LSP27TheGrid = [
+  {
+    type: GridWidgetType.TITLE_LINK,
+    width: 1,
+    height: 2,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.TEXT,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.IFRAME,
+    width: 1,
+    height: 2,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.IMAGE,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+]
+
+const TEST_GRID_VARIANT_1: LSP27TheGrid = [
+  {
+    type: GridWidgetType.IMAGE,
+    width: 2,
+    height: 2,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.TEXT,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.IMAGE,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+]
+
+const TEST_GRID_VARIANT_2: LSP27TheGrid = [
+  {
+    type: GridWidgetType.IMAGE,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.IFRAME,
+    width: 2,
+    height: 2,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.TEXT,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.IMAGE,
+    width: 2,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+]
+
+const TEST_GRID_VARIANT_3: LSP27TheGrid = [
+  {
+    type: GridWidgetType.TITLE_LINK,
+    width: 1,
+    height: 2,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.TEXT,
+    width: 2,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+]
+
+const TEST_GRID_VARIANT_4: LSP27TheGrid = [
+  {
+    type: GridWidgetType.IFRAME,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.IMAGE,
+    width: 1,
+    height: 2,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.TEXT,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+  {
+    type: GridWidgetType.IMAGE,
+    width: 1,
+    height: 1,
+    properties: { prop1: 'value1', prop2: 'value2' },
+  },
+]
+
+describe('toGridLayoutItems', () => {
+  beforeEach(() => {
+    // jest.resetModules();
+  })
+
+  it.each([
```

Comment:
It's very hard to read such tests. It would be better if we have describe blocks per each widget.

Reply:
👍  Ill make the tests clearer